src/ArborX_LinearBVH.hpp

```diff
@@ -239,15 +239,16 @@ void BasicBoundingVolumeHierarchy<MemorySpace, BoundingVolume, Enable>::query(
 
   using Tag = typename Details::AccessTraitsHelper<Access>::tag;
   std::string profiling_prefix = "ArborX::BVH::query";
-  if (std::is_same<Tag, Details::SpatialPredicateTag>{})
+  if constexpr (std::is_same_v<Tag, Details::SpatialPredicateTag>)
```

Comment:
Do not put unrelated (somewhat unnecessary) changes in the same commit.
Either revert or move into another commit with an appropriate message.
(And if the rational is more consistency, you might want to decide where you want the `::`, in the prefix to like here in what gets appended to it.